test/Microsoft.IdentityModel.Protocols.OpenIdConnect.Tests/ConfigurationManagerTests.cs

```diff
@@ -176,20 +176,37 @@ public void GetConfiguration()
             if (!object.ReferenceEquals(configuration, configuration2))
                 context.Diffs.Add("!object.ReferenceEquals(configuration, configuration2) (3)");
 
-            // Refresh should have no effect
             configManager = new ConfigurationManager<OpenIdConnectConfiguration>("OpenIdConnectMetadata.json", new OpenIdConnectConfigurationRetriever(), docRetriever);
             configuration = configManager.GetConfigurationAsync().Result;
-            configManager.RefreshInterval = TimeSpan.FromHours(10);
+            // First force refresh should pickup new config
             configManager.RequestRefresh();
+            TestUtilities.SetField(configManager, "_metadataAddress", "OpenIdConnectMetadata2.json");
             configuration2 = configManager.GetConfigurationAsync().Result;
-            IdentityComparer.AreEqual(configuration, configuration2, context);
-            if (!object.ReferenceEquals(configuration, configuration2))
-                context.Diffs.Add("!object.ReferenceEquals(configuration, configuration2) (4)");
+            if (IdentityComparer.AreEqual(configuration, configuration2))
+                context.Diffs.Add("IdentityComparer.AreEqual(configuration, configuration2), should be different");
+            if (object.ReferenceEquals(configuration, configuration2))
+                context.Diffs.Add("object.ReferenceEquals(configuration, configuration2) (4)");
+            // Next force refresh shouldn't pickup new config, as RefreshInterval hasn't passed
+            configManager.RequestRefresh();
+            TestUtilities.SetField(configManager, "_metadataAddress", "OpenIdConnectMetadata.json");
+            var configuration3 = configManager.GetConfigurationAsync().Result;
+            IdentityComparer.AreEqual(configuration2, configuration3, context);
+            if (!object.ReferenceEquals(configuration2, configuration3))
+                context.Diffs.Add("!object.ReferenceEquals(configuration2, configuration3) (5)");
+            // Next force refresh should pickup config since, RefreshInterval is set to 1s
+            configManager.RefreshInterval = TimeSpan.FromSeconds(1);
+            Thread.Sleep(2000);
```

Comment:
refresh interval is set to 1ms here, adding sleep for 2ms because with the jitter max it can be doubled.